frontend/packages/console-dynamic-plugin-sdk/src/api/common-types.ts

```diff
@@ -11,7 +11,8 @@ export type ExtensionHookResult<T> = [T, boolean, any];
 export type ExtensionK8sModel = {
   group: string;
   version: string;
-  kind: string;
+  kind?: string;
+  resource?: string;
```

Comment:
This type is shared across the console, so let us create our own type for pinned resources instead of changing this. Other extensions like "feature flags", "yaml templates" that uses this types would not expect that there are models without a kind.